internal/service/eks/addon_test.go

```diff
@@ -246,6 +246,38 @@ func TestAccEKSAddon_serviceAccountRoleARN(t *testing.T) {
 	})
 }
 
+func TestAccEKSAddon_configurationValues(t *testing.T) {
+	var addon eks.Addon
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_eks_addon.test"
+	configurationValues := "{\"env\": {\"WARM_ENI_TARGET\":\"2\",\"ENABLE_POD_ENI\":\"true\"},\"resources\": {\"limits\":{\"cpu\":\"100m\",\"memory\":\"100Mi\"},\"requests\":{\"cpu\":\"100m\",\"memory\":\"100Mi\"}}}"
+	addonName := "vpc-cni"
+	addonVersion := "v1.10.4-eksbuild.1"
+	ctx := context.Background()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t); testAccPreCheck(t); testAccPreCheckAddon(t) },
+		ErrorCheck:               acctest.ErrorCheck(t, eks.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckAddonDestroy,
+		Steps: []resource.TestStep{
```

Comment:
Since ConfigurationValues are a primary interface path for customers, and they are likely to mutate them for needs such as more memory or more changing env_vars, we should include a multi-step AccTest to verify not only the setting of the configuration values, but also the updating of configuration values.

Reply:
Added updates to the config values with multiple steps